Include/py_curses.h

```diff
@@ -23,7 +23,7 @@
 # endif
 #endif
 
-#if !defined(HAVE_CURSES_IS_PAD) && defined(WINDOW_HAS_FLAGS)
```

Comment:
It's annoying that all platforms now inspect structure members, only to working a bug specific to macOS. Maybe only ignore HAVE_CURSES_IS_PAD on macOS?

Reply:
Please change the test to at least only make this change on macOS.

Reply:
Hi, @sorcio 
I'd suggest this #if block to put inside first `__APPLE__` condition to make more clearly understand the effect only for macOS. (it changed by me at #4164)